dotcom-rendering/src/web/components/ShareIcons.tsx

```diff
@@ -67,6 +67,14 @@ const decideIconColor = (format: ArticleFormat, context: Context) => {
 			fill: ${palette.fill.shareIconGrayBackground};
 		`;
 	}
+	if (
+		format.design === ArticleDesign.NewsletterSignup &&
+		context === 'ArticleMeta'
```

Comment:
If design is equal to NewsletterSignup then I think this context value is always going to be `ArticleMeta`?

If so, I think we can move this decision logic into `decidePalette`. 

### `decidePalette`? What is this?
This is where we make decisions about what colour to use for things in DCR. We built this (big, long) abstraction as a way to centralise where decisions are made about colour based on `format`. It is designed to try and make this kind of logic easier to reason about and more scalable. We often get requests and features to edit this kind of code so it's nice to have a structure to it.